### PR TITLE
fix(rest) : Move GenerateLicenseInfoFile rest end point to SW360reportcontroller

### LINKED ISSUE #2441: fix(rest) : Move GenerateLicenseInfoFile rest end point to SW360reportcontroller.
### Description

This bug is creating to Move GenerateLicenseInfoFile rest end point to Sw360Reportcontroller, so we can align all the reports in one single path. 

#### How to reproduce

Check in the main branch before this PR get merge, in the Rest API Docs, test the **Download License Info** and then cross verify the functionality with associated PR with this issue, both should behave same way.

### PR DESCRIPTION
**Pull request by keerthi-bl**
### Description
Issue reference 
Closes #2441 

#### How to reproduce

Sample test : 
URL : http://localhost:8080/resource/api/reports?module=licenseInfo&projectId=8fef50eafc484e33bb8ebb33618ed56d&generatorClassName=DocxGenerator&variant=DISCLOSURE

Result : 

image

Can download from rest!

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.

image

**Comment by nikkuma7** [comment]
[LicenseInfo-6thOblication-2024-05-10_06_18_51.docx](https://github.com/eclipse-sw360/sw360/files/15271435/LicenseInfo-6thOblication-2024-05-10_06_18_51.docx)

**Comment by GMishx** [comment]
@keerthi-bl please rebase with latest main to resolve merge conflict.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java [hunk 1]
```diff
@@ -5,28 +5,77 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import javax.servlet.http.HttpServletRequest;
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java:22]:
```suggestion
import jakarta.servlet.http.HttpServletRequest;
```

**Review by GMishx — CHANGES_REQUESTED** [review]
Build is failing. Rest looks good.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**